### PR TITLE
MNT: Begin Issues

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Increase the begin timeout to 15s
- Add more pointed error messages to the wait timeouts

## Motivation and Context

First pass at addressing #83 

## How Has This Been Tested?

Added simple coverage tests for the new exceptions

### CODE REVIEW

File: pcdsdaq/daq.py [hunk 1]
```diff
@@ -222,7 +227,12 @@ def wait(self, timeout=None):
         if self.state == 'Running':
             if self._events or self._duration:
                 status = self._get_end_status()
-                status_wait(status, timeout=timeout)
+                try:
+                    status.wait(timeout=timeout)
+                except (StatusTimeoutError, WaitTimeoutError):
+                    msg = ('Timeout after %s seconds waiting for daq to finish '
+                           'acquiring.')
```
Comment (klauer) [inline pcdsdaq/daq.py:234]:
```suggestion
                           'acquiring.' % timeout)
```

File: pcdsdaq/daq.py [hunk 2]
```diff
@@ -282,7 +292,12 @@ def begin(self, events=_CONFIG_VAL, duration=_CONFIG_VAL,
                 self.preconfig(record=record, show_queued_cfg=False)
             begin_status = self.kickoff(events=events, duration=duration,
                                         use_l3t=use_l3t, controls=controls)
-            begin_status.wait(timeout=self._begin_timeout)
+            try:
+                begin_status.wait(timeout=self._begin_timeout)
+            except (StatusTimeoutError, WaitTimeoutError):
+                msg = 'Timeout after %s seconds waiting for daq to begin.'
+                raise DaqTimeoutError(msg, self._begin_timeout) from None
```
Comment (klauer) [inline pcdsdaq/daq.py:299]: Ditto

File: pcdsdaq/sim/pydaq.py [hunk 3]
```diff
@@ -117,6 +118,10 @@ def begin(self, *, events=None, l1t_events=None, l3t_events=None,
             self._done_flag.clear()
             if self._record:
                 Control._run_number += 1
+            if self._begin_delay:
+                delay = self._begin_delay
+                self._begin_delay = 0
+                time.sleep(delay)
```
Comment (klauer) [inline pcdsdaq/sim/pydaq.py:124]: Not sure if it makes any difference (or sense here), but could the sleep happen in `begin_thread` instead?

**Review by klauer — APPROVED** [review]
A couple fixes needed, but seems fine otherwise.

File: pcdsdaq/daq.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsdaq/daq.py:234]: No, this is not correct. The % timeout happens in the logger call.

File: pcdsdaq/sim/pydaq.py (hunk 3 above)
Comment (ZLLentz) [reply pcdsdaq/sim/pydaq.py:124]: This goal here is to simulate the real behavior. The `pydaq` module pauses and does not return from `begin` until the daq actually starts running, which can take an indeterminate time. This is the call that is timing out in prod with a timeout of 3 seconds.

File: pcdsdaq/daq.py (hunk 1 above)
Comment (klauer) [reply pcdsdaq/daq.py:234]: Oh, ok. I'll have to look into the details to see who's formatting this string, but it looked wrong from a context-free PR.

Comment (ZLLentz) [reply pcdsdaq/daq.py:234]: Let me quickly double-check these in an interactive session because brain is paranoia now

Comment (ZLLentz) [reply pcdsdaq/daq.py:234]: Ok I had a massive brain fart and thought I was passing into a logger, but I was passing into an Exception

Comment (ZLLentz) [reply pcdsdaq/daq.py:234]: I can only blame myself, but I can also blame python for having at least three different ways to format strings

Comment (klauer) [reply pcdsdaq/daq.py:234]: Ahh, no big deal 🤷‍♂️ no need for blame. Let's catch/fix/merge and move onto the next issue in the overflowing queue...

**Comment by klauer** [comment]
LGTM; merge at will.

**Pull request merged**